cdc/sinkv2/eventsink/mq/worker.go

```diff
@@ -59,8 +60,15 @@ type worker struct {
 	msgChan *chann.Chann[mqEvent]
 	// ticker used to force flush the messages when the interval is reached.
 	ticker *time.Ticker
+
+	// builder is used to build encoders
+	builder codec.EncoderBuilder
 	// encoder is used to encode the messages.
 	encoder codec.EventBatchEncoder
+	// encoderGroup only support non batch encode protocol at the moment.
+	encoderGroup       codec.EncoderGroup
+	encoderConcurrency int
```

Comment:
Why do we need to keep it on struct? It seems needless because we already have the group.

Reply:
`encoderGroup` is initialized in the `run` method only for non-batch protocol at the moment.

`encoderConcurrency` can be passed to the `run` method explicitly or by the current way.

I will make encoder group support batched protocols, after that we can initialize the encoder group in the `newWorker` method, after that we can just keep one `encoderBuilder` field in the worker struct, and remove others.